docs/source/examples/loschmidt_echo_revival_zne.md

```diff
@@ -0,0 +1,428 @@
+                   scale_factor: float = None) -> float:
+    # Transpile the circuit
+    backend = QasmSimulator(noise_model=noise_model)
+    exec_circuit = transpile(
+        circuit,
+        backend=backend,
+        basis_gates=["u1", "u2", "u3", "cx"],
+        optimization_level=0
+    )
+
+    # Apply gate folding
+    folded_circuit = exec_circuit if scale_factor is None \
+                     else fold_gates_at_random(exec_circuit, scale_factor)
+    
+    job = backend.run(folded_circuit, shots=n_shots)
+
+    # Get the probability of returning to $\ket{0^{\otimes N}}$
+    counts = job.result().get_counts()
+    ket = "0" * circuit.num_qubits
+    if ket in counts:
+        return counts[ket]/n_shots
+    return 0.0
+
+def run_depolarizing_noise(params : SimulationParams = SimulationParams(),
+                           noise_level : float = 0.001,
+                           scale_factor : float = None) -> tuple[np.ndarray[float]]:
+    basis_gates = ["u1", "u2", "u3", "cx"]
+    noise_model = NoiseModel(basis_gates)
+    # Add depolarizing noise to the 1-qubit gates in the basis set
+    noise_model.add_all_qubit_quantum_error(
+        depolarizing_error(noise_level, 1), basis_gates[:3]
+    )
+    
+    echo = np.array([
+        simulate_noisy(
+            create_circuit(params.dt, k, n_qubits=params.n_qubits, measure=True),
+            noise_model,
+            params.n_shots,
+            scale_factor
+        )
+        for k in range(0, params.M + 1)
+    ])    
+
+    return params.t_vals, echo
+```
+
+```{code-cell} ipython3
+low_noise = 0.0005
+result_depolarizing = run_depolarizing_noise(noise_level=low_noise)
+```
+
+```{code-cell} ipython3
+setup_plot("Ideal and noisy simulations")
+legend = []
+add_to_plot(*result_ideal, "ideal", legend)
+add_to_plot(
+    *result_depolarizing,
+    "depolarizing noise, $p = {}$".format(low_noise),
+    legend)
+plt.legend(legend)
+plt.show()
+```
+
+As expected, the Loschmidt echo revival is weaker than in the ideal cases, and we get a lower peak. Applying gate folding suppresses the peak further.
+
+```{code-cell} ipython3
+scale_factors = [1, 2, 3]
+result_depolarizing_scaled = [
+    run_depolarizing_noise(
+        noise_level=low_noise,
+        scale_factor=alpha)
+    for alpha in scale_factors
+]
+```
+
+```{code-cell} ipython3
+setup_plot("Scaled noisy simulations")
+legend = []
+add_to_plot(*result_ideal, "ideal", legend)
+for alpha, result in zip(scale_factors, result_depolarizing_scaled):
+    add_to_plot(*result, "$\\alpha = {}$".format(alpha), legend)
+plt.legend(legend)
+plt.show()
+```
+
+## Error mitigation with zero-noise extrapolation
+At this level of noise, we can use ZNE to mostly recover the ideal result. Running the noisy simulation is expensive (especially as the scale factor $\alpha$ increases), so rather than using the high-level functions for ZNE, we apply the static method {meth}`.RichardsonFactory.extrapolate` to the results from the previous cell.
```

Comment:
Smart! And great to demonstrate this workflow since I don't think we walk through this as nicely as you do here.